src/megatron/bridge/models/kimi/kimi_provider.py

```diff
@@ -0,0 +1,139 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from dataclasses import dataclass, field
+from functools import partial
+from typing import TYPE_CHECKING, Callable, List, Optional, Union
+
+import torch
+import torch.nn.functional as F
+from megatron.core.models.gpt.gpt_layer_specs import get_gpt_decoder_block_spec
+from megatron.core.transformer.transformer_config import MLATransformerConfig
```

Comment:
after #627 we extend MLATransformerConfig so that we can initialize the config, apply any overrides, and then run the post_init check once in the training loop. otherwise the post init might initialize some dependent values that may not be updated when the overrides are applied, even if post_init is explicitly called again

```suggestion
from megatron.bridge.models.transformer_config import MLATransformerConfig
```